### ISSUE TITLE
Email send but not recived

### ISSUE THREAD

**Issue by DWS-paris** [mailjet/mailjet-apiv3-nodejs#207]
Hello!

I try from couple of hours to send email within Mailjet. I'd create a API with Express and a specific POST route to call the following function:

```js
/*
    [Imports] Node.js
    Import NPM modules
*/
    const Mailjet = require('node-mailjet'); // https://www.npmjs.com/package/node-mailjet
//

/* 
    [SERVICE] Function to send email with Node Mailer / Mailjet
    @param{transporter}: used transporter
    @param{from}: sender email STRING
    @param{to}: Email address STRING
    @param{topic}: Email subject STRING
    @param{content}: Email HTML body STRING

    => node-mailjet: 4.0.1
    => node: 16.15.1
    => express: 4.18.1
    => pnpm: 7.1.9
*/
    const sendEmail = async (to, topic, content) => {
        return new Promise( async (resolve, reject) => {
            /* 
                [MAILER] Transporter
                Setup Mailjet transporter
            */
                const mailjetTransporter = new Mailjet({
                    apiKey: process.env.MAILJET_API_KEY,
                    apiSecret: process.env.MAILJET_API_SECRET,
                });
            //
            
            /* 
                [SEND] Email
                Use Mailjet transporter to send email
            */
                mailjetTransporter
                .post('send', { version: 'v3.1' })
                .request({
                    Messages: [
                        {
                            From: {
                                Email: process.env.SMTP_FROM_EMAIL,
                                Name: process.env.SMTP_FROM_NAME
                            },
                            To: to,
                            Subject: topic,
                            TextPart: topic,
                            HTMLPart: content
                        }
                    ]
                })
                .then( mailjetSuccessResponse => resolve(mailjetSuccessResponse) )
                .catch( mailjetErrorResponse => reject(mailjetErrorResponse) );
            //
        })
    }
//
```

When I try this function in Postman I've got a success response with the following JSON object: 

```json
"response": {
    "req": {
        "method": "POST",
        "url": "https://api.mailjet.com/v3.1/send",
        "data": {
            "Messages": [
                {
                    "From": {
                        "Email": "contact@xxx.com",
                        "Name": "Company"
                    },
                    "To": [
                        {
                            "Email": "xxx@gmail.com",
                            "Name": "John Doe"
                        }
                    ],
                    "Subject": "[DEBUG] Mailjet test",
                    "TextPart": "[DEBUG] Mailjet test",
                    "HTMLPart": "<h1>Hello</h1><p>this is a test</p>"
                }
            ]
        },
        "headers": {
            "user-agent": "mailjet-api-v3-nodejs/4.0.0",
            "content-type": "application/json",
            "authorization": "Basic {{AUTH-KEY}}",
            "accept": "application/json"
        }
    },
    "header": {
        "content-length": "267",
        "content-type": "application/json; charset=UTF-8",
        "x-mj-request-guid": "f16f0749-2d9b-4aed-8755-02800ba6c896",
        "date": "Tue, 21 Jun 2022 09:49:16 GMT",
        "connection": "close"
    },
    "status": 200,
    "text": "{\"Messages\":[{\"Status\":\"success\",\"CustomID\":\"\",\"To\":[{\"Email\":\"xxx@gmail.com\",\"MessageUUID\":\"37093fab-6beb-4533-9029-dd061c3db261\",\"MessageID\":288230388960990212,\"MessageHref\":\"https://api.mailjet.com/v3/REST/message/288230388960990212\"}],\"Cc\":[],\"Bcc\":[]}]}"
},
"body": {
    "Messages": [
        {
            "Status": "success",
            "CustomID": "",
            "To": [
                {
                    "Email": "xxx@gmail.com",
                    "MessageUUID": "37093fab-6beb-4533-9029-dd061c3db261",
                    "MessageID": "288230388960990212",
                    "MessageHref": "https://api.mailjet.com/v3/REST/message/288230388960990212"
                }
            ],
            "Cc": [],
            "Bcc": []
        }
    ]
}
```

But when I try to access `https://api.mailjet.com/v3/REST/message/288230388960990212` I always have: 

```json
{ 
    "ErrorInfo" : "", 
    "ErrorMessage" : 
    "Object not found", 
    "StatusCode" : 404 
}
```

I don't find so much documentation about this problem, could you please give me a hand?

**Comment by scroll17** [comment]
Hey @DWS-paris 

I tried to do the same request as you use this code: 
```javascript
const sendEmail = async (to, topic, content) => new Promise((resolve, reject) => {
  const mailjetTransporter = new Mailjet({
    apiKey: process.env.MJ_APIKEY_PUBLIC,
    apiSecret: process.env.MJ_APIKEY_PRIVATE,
  });

  mailjetTransporter
    .post('send', { version: 'v3.1' })
    .request({
      Messages: [
        {
          From: {
            Email: process.env.SMTP_FROM_EMAIL,
            Name: process.env.SMTP_FROM_NAME,
          },
          To: to,
          Subject: topic,
          TextPart: topic,
          HTMLPart: content,
        },
      ],
    })
    .then((mailjetSuccessResponse) => resolve(mailjetSuccessResponse))
    .catch((mailjetErrorResponse) => reject(mailjetErrorResponse));
});

(async () => {
  try {
    const result = await sendEmail(
      [
        {
          Email: 'pilot@mailjet.com',
          Name: 'Pilot',
        },
      ],
      '[DEBUG] Mailjet test',
      '<h1>Hello</h1><p>this is a test</p>',
    );

    console.log('res => ', result);
    console.dir(result.body, { depth: 10 });
  } catch (e) {
    console.log('e => ', e);
  }
})();
```

But I receive a response with the correct `MessageHref` and `MessageID`: 
```javascript
[Object: null prototype] {
  Messages: [
    [Object: null prototype] {
      Status: 'success',
      CustomID: '',
      To: [
        [Object: null prototype] {
          Email: 'pilot@mailjet.com',
          MessageUUID: 'f71c549a-e85a-4a56-a14d-0d82005d609f',
          MessageID: '576460765131762352',
          MessageHref: 'https://api.mailjet.com/v3/REST/message/576460765131762352'
        }
      ],
      Cc: [],
      Bcc: []
    }
  ]
}
```

And when I follow this link I receive the correct response for `MessageID`.

---
You can try to use this code snippet to retrieve a list of messages which sent in the **passed time**: 
```javascript
(async () => {
  try {
    const mailjetTransporter = new Mailjet({
      apiKey: process.env.MJ_APIKEY_PUBLIC,
      apiSecret: process.env.MJ_APIKEY_PRIVATE,
    });

    const targetUnixTine = new Date().valueOf() - (1000 * 60 * 27); // 27m

    const result = await mailjetTransporter
      .get('message')
      .request({
        FromTS: new Date(targetUnixTine).toISOString(),
      });

    console.log('res => ', result);
    console.dir(result.body, { depth: 10 });
  } catch (e) {
    console.log('e => ', e);
  }
})();
``` 

As well you can visit [this page](https://dev.mailjet.com/email/reference/messages/#v3_get_message) where you can find more query params that can help you find your message.

---
In summary: 
I think it's can be a bug on the **Mailjet** server-side because I can't reproduce it on my local host.
Please follow this [link](https://app.mailjet.com/support) where you can **open a ticket**.

**Comment by scroll17** [comment]
This issue will be closed
